dealership/vehicles.py

```diff
@@ -1,15 +1,51 @@
 class Vehicle(object):
     def __init__(self, maker, model, year, base_price, miles):
-        pass
+        self.maker = maker
+        self.model = model
+        self.year = year
+        self.base_price = base_price
+        self.miles = miles
+        
+    def get_sale_multiplier(self):
+        raise NotImplementedError()
+        
+    def get_purchase_multiplier(self):
+        raise NotImplementedError()
+    
+    def sale_price(self):
+        return self.base_price * self.get_sale_multiplier()
 
+    def purchase_price(self):
+        return self.sale_price() - (self.get_purchase_multiplier() * self.miles)
 
-class Car(Vehicle):
-    pass
 
+class Car(Vehicle):
+    def get_sale_multiplier(self):
+        return 1.2
+    def get_purchase_multiplier(self):
+        return 0.004
+    def get_interest_rate(self):
+        return 1.07
+    def get_lease_multiplier(self):
+        return 1.2
 
 class Motorcycle(Vehicle):
-    pass
-
+    def get_sale_multiplier(self):
+        return 1.1
+    def get_purchase_multiplier(self):
+        return 0.009
+    def get_interest_rate(self):
+        return 1.03
+    def get_lease_multiplier(self):
+        return 1.
 
 class Truck(Vehicle):
-    pass
+    def get_sale_multiplier(self):
+        return 1.6
+    def get_purchase_multiplier(self):
+        return 0.02
+    def get_interest_rate(self):
+        return 1.11
+    def get_lease_multiplier(self):
+        return 1.7
```

Comment:
This is an interesting way to implement the multipliers! An alternative might be to use constants (expressed as a variable in uppercase letters), set them in the superclass and override them in the subclasses' `__init__`, because you want to indicate with your syntax that this number is not the result of computation, but instead it's a number that doesn't change at any moment in time.